api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -581,6 +581,32 @@ message HttpConnectionManager {
   // *not* the deprecated but similarly named :ref:`stream_error_on_invalid_http_messaging
   // <envoy_v3_api_field_config.core.v3.Http2ProtocolOptions.stream_error_on_invalid_http_messaging>`
   google.protobuf.BoolValue stream_error_on_invalid_http_message = 40;
+
+  // The configuration of the original-IP detection extension.
+  //
+  // If not set, Envoy uses the default remote IP detection mechanism.
+  //
+  // When the remote address of the downstream connection is not used as
+  // the detection mechanism for the request's original address, Envoy
+  // will rely on the :ref:`config_http_conn_man_headers_x-forwarded-for` HTTP
+  // header to detect the address.
+  //
+  // However, relying on the :ref:`config_http_conn_man_headers_x-forwarded-for`
+  // header doesn't necessarily work for all network configurations. For instance,
+  // if the number of trusted hops isn't consistent for requests flowing into
+  // Envoy the header cannot be used reliably. For such cases, the following
+  // option allows for a custom extension to be provided.
+  //
+  // If configured, the custom extension will be called along with the
+  // request headers and the downstream directly connected address when the
```

Comment:
I'd be inclined to say request headers and information about the downstream connection, such as direct connected address, so we don't have to update API every time we add info.